src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -136,10 +136,19 @@ public void RequestEnd(HttpContext httpContext, Exception exception, HostingAppl
                 StopActivity(httpContext, activity, context.HasDiagnosticListener);
             }
 
-            if (context.EventLogEnabled && exception != null)
+            if (context.EventLogEnabled)
             {
-                // Non-inline
-                HostingEventSource.Log.UnhandledException();
+                if (exception != null)
+                {
+                    // Non-inline
+                    HostingEventSource.Log.UnhandledException();
```

Comment:
Is there a reason we don't put exception details in the message?

Reply:
1. This isn't new code
2. You can't pass the exception itself (that isn't supported in EventSource writes), you could pass the text though.

Reply:
> 1. This isn't new code

Yes, I am aware.